docs/README.md

```diff
@@ -1,3 +1,201 @@
-# Documentation
+# .NET Swift interop tooling documentation
 
-This document outlines the functional design of the tooling and projections between Swift and .NET.
\ No newline at end of file
+This document provides a detailed overview of the .NET Swift interop tooling, focusing on the projections between Swift and .NET, and the functional design of the tooling.
+
+## Projections
+
+This section outlines the mappings between Swift and .NET types, and describes how the tool generates C# bindings for a Swift library. For types with similar semantics across Swift and .NET, direct interop is possible and bindings are generated. For types without direct projection, additional Swift wrappers are required and it is the user's responsibility to generate these wrappers. At this stage, the tool is designed to avoid generating any Swift code if possible to avoid complexity of maintaining different Swift compiler and SDK versions. The tool should only generate bindings for Swift types that are part of the stable ABI and don't evolve.
+
+### Primitive types
+
+The table below lists the Swift types and their corresponding C# types.
+
+| Swift Type                      | C# Type  |
+| ------------------------------- | -------- |
+| `Swift.Int64`                   | `long`   |
+| `Swift.UInt64`                  | `ulong`  |
+| `Swift.Int32`                   | `int`    |
+| `Swift.UInt32`                  | `uint`   |
+| `Swift.Int16`                   | `short`  |
+| `Swift.UInt16`                  | `ushort` |
+| `Swift.Int8`                    | `sbyte`  |
+| `Swift.UInt8`                   | `byte`   |
+| `Swift.UnsafeRawPointer`        | `IntPtr` |
```

Comment:
The `IntPtr` / `UIntPtr` are aliases to `nint` / `nuint`. We should be using actual pointers like `void*` in C# when expressing the pointer concept.

Reply:
I agree with Aaron here. We should map `UnsafeRawPointer` to `void*`.

Reply:
Ok, thanks. Added to the implementation plan in https://github.com/dotnet/runtimelab/pull/2512.